office-backend/src/main/kotlin/flock/community/office/monitoring/backend/domain/service/WeatherService.kt

```diff
@@ -0,0 +1,110 @@
+package flock.community.office.monitoring.backend.domain.service
+
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.context.properties.ConfigurationProperties
+import org.springframework.boot.context.properties.ConstructorBinding
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import org.springframework.http.HttpHeaders
+import org.springframework.http.MediaType
+import org.springframework.stereotype.Service
+import org.springframework.web.reactive.function.client.WebClient
+import reactor.core.publisher.Mono
+
+@Service
+class WeatherService(
+    private val webClient: WebClient,
+    private val openWeatherMapConfig: OpenWeatherMapConfig
+) {
+
+    private val flockOfficeCoordinates: Coord = Coord(52.09266175027509, 5.122345051397365)
+
+    private fun buildUrl(): String =
+        "?lat=${flockOfficeCoordinates.lat}&lon=${flockOfficeCoordinates.lon}&appid=${openWeatherMapConfig.apiKey}&units=metric"
+
+    fun getPrediction(): Mono<WeatherPrediction> {
```

Comment:
De WebClient gebruikt Reactor maar dat wil je eigenlijk niet naar de consumer van deze service uitlekken (tenzij we daar misschien ook Reactor zouden gebruiken maar daar hebben we al Coroutines). Het lijkt me daarom beter dat dit een `suspend fun` is en dat die gewoon een `WeatherPrediction` teruggeeft. 

Er zijn Coroutine extension functions om van een Mono of Flux naar een Coroutine te gaan. Je kan `.awaitBody<WeatherPrediction::class.java>()` gebruiken als laatste regel in deze functie.

Meer info onder het stukje: "Asynchronous operations with imperative code":
https://spring.io/blog/2019/04/12/going-reactive-with-spring-coroutines-and-kotlin-flow

Reply:
In plaats van bodyToMono moet je die dan hebben denk ik

Reply:
Het heeft even geduurd, maar de service returned nu een simpel WeatherPrediction object ipv een mono. De test is ook aangepast en het werkt :)